vispy/visuals/visual.py

```diff
@@ -678,3 +679,76 @@ def __init__(self, visual):
         for filt in self._vshare.filters:
             for v in self._subvisuals:
                 filt._attach(v)
+
+
+class updating_property:
+    """A property descriptor that autoupdates the Visual during attribute setting.
+
+    Use this as a decorator in place of the @property when you want the attribute to trigger
+    an immediate update to the visual upon change.
+    For example, the following code examples are equivalent:
+
+    class SomeVisual1:
```

Comment:
Could you update these examples to base themselves off of the BaseVisual class (or whatever the actual name is)? Just in case in the future this class depends more on actual BaseVisual functionality we know it is documented here. Thoughts?

Reply:
yeah, can do.  I'm actually reviewing the inheritance chain for Visuals and how they draw themselves right now... cause I'm still not 100% sure just how generally this decorator actually is.  (That is, I don't know how widely used that `_need_update` and `.update()` pattern is in the Visuals... and how many other visuals would actually use this.).  Once I do, I'll try to update with a few more modular commits that implement this pattern in a couple more Visuals.

Reply:
actually... can you help me on that?  It's still not clear to me why `Visual.update()` even does anything.  From what I have found, it will ultimately call `BaseVisual.update()` which simply emits the update event (`BaseVisual.events.update()`)... but I am unable to find anywhere in vispy where that event is actually connected (except in `CompoundVisual`, where it simply is connected to calling update on the subvisuals).  How does `BaseVisual.update()` eventually trigger the program to draw?

Reply:
Give me a bit and I'll try to find the parts you're looking for, but I would keep this PR simple with only the necessary changes (don't change other visuals).

Reply:
So without looking too much harder (I always forget the exact calling sequence), the individual visual `update` calls should trigger the call on `Canvas.update` which ends up calling `_vispy_update` on the backend. For some backends like Qt this is just a redraw (which I think is also called `update` in Qt).

I can't find the connection between the update event and the canvas update either...

Reply:
Ok, I feel a little less dumb now. When working with a SceneCanvas the important piece of code is the `Node` base class which is used for all `vispy.scene.visuals` versions of the Visual objects:

https://github.com/vispy/vispy/blob/9dbe751dcdbaaa1d17d0a5885f15872b9b0741a8/vispy/scene/node.py#L325-L333

So that's what gets called when visuals do `self.update` on a Visual in a SceneCanvas. In a normal `Canvas` situation though there is nothing that automatically connects `visual.events.update` to any other methods. I found examples that did this themselves (connect the update event to Canvas.update).

So in summary, `Node` overrides the default `update` method for the Visual objects.

Reply:
aha!  thanks!  

OK, I won't make any changes to other visuals in this PR, but I do want to just look around a bit to make sure that this isn't introducing a whole new API for creating "updatable" visuals that really doesn't apply very far beyond the `AxisVisual`.